From 916ed65b3901731fff02eceadcbc5b18f4c27d6f Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Sat, 14 Jul 2018 09:40:37 +0200
Subject: [PATCH] Remove unused variables from configure.ac.

Also replace
  configure.ac: NO_PREFIX_PACKAGE_FOO_DIR = "share/foo"
  configure.ac: AC_SUBST(NO_PREFIX_PACKAGE_FOO_DIR)
and
  Makefile.am : .. = $(prefix)/@NO_PREFIX_PACKAGE_FOO_DIR@
with
  Makefile.am : .. = $(datadir)/foo
for simplicity and conformity with the usual meaning of datadir.

These changes may require regeneration by Anjuta.
---
 Makefile.am       |  2 +-
 configure.ac      | 85 -----------------------------------------------
 icons/Makefile.am |  2 +-
 3 files changed, 2 insertions(+), 87 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index db5ec16..d61162f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,7 +27,7 @@ DISTCLEANFILES = \
 	m4/* \
 	po/Makefile.in.in
 
-gnomemenudir = $(prefix)/@NO_PREFIX_PACKAGE_MENU_DIR@
+gnomemenudir = $(datadir)/applications
 gnomemenu_DATA = org.nongnu.lingot.desktop
 appdatadir = $(datadir)/metainfo
 appdata_DATA = org.nongnu.lingot.appdata.xml
diff --git a/configure.ac b/configure.ac
index 08f208f..cfb1259 100644
--- a/configure.ac
+++ b/configure.ac
@@ -162,90 +162,6 @@ if test "x$uselibfftw" = "xyes"; then
  fi
 fi
 
-
-dnl PKG_CHECK_MODULES([GLADE], [libglade-2.0])
-dnl AC_SUBST(GLADE_CFLAGS)
-dnl AC_SUBST(GLADE_LIBS)
-
-dnl Set PACKAGE_LOCALE_DIR in config.h.
-if test "x${prefix}" = "xNONE"; then
-  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "${ac_default_prefix}/${DATADIRNAME}/locale", [Package local directory])
-else
-  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "${prefix}/${DATADIRNAME}/locale", [Package local directory])
-fi
-
-dnl Set PACKAGE SOURCE DIR in config.h.
-packagesrcdir=`cd $srcdir && pwd`
-
-dnl Set PACKAGE PREFIX
-if test "x${prefix}" = "xNONE"; then
-  packageprefix=${ac_default_prefix}
-else
-  packageprefix=${prefix}
-fi
-
-dnl Set PACKAGE DATA & DOC DIR
-packagedatadir=share
-packagedocdir=doc/${PACKAGE}
-
-dnl Set PACKAGE DIRs in config.h.
-packagepixmapsdir=share/pixmaps/${PACKAGE}
-packageicondir=share/icons/hicolor/scalable/apps/
-packagehelpdir=share/gnome/help/${PACKAGE}
-packagemenudir=share/applications
-
-dnl Subst PACKAGE_DATA_DIR.
-NO_PREFIX_PACKAGE_DATA_DIR="${packagedatadir}"
-AC_SUBST(NO_PREFIX_PACKAGE_DATA_DIR)
-PACKAGE_DATA_DIR="${packageprefix}/${packagedatadir}"
-AC_SUBST(PACKAGE_DATA_DIR)
-
-dnl Subst PACKAGE_DOC_DIR.
-NO_PREFIX_PACKAGE_DOC_DIR="${packagedocdir}"
-AC_SUBST(NO_PREFIX_PACKAGE_DOC_DIR)
-PACKAGE_DOC_DIR="${packageprefix}/${packagedocdir}"
-AC_SUBST(PACKAGE_DOC_DIR)
-
-dnl Subst PACKAGE_PIXMAPS_DIR.
-NO_PREFIX_PACKAGE_PIXMAPS_DIR="${packagepixmapsdir}"
-AC_SUBST(NO_PREFIX_PACKAGE_PIXMAPS_DIR)
-PACKAGE_PIXMAPS_DIR="${packageprefix}/${packagepixmapsdir}"
-AC_SUBST(PACKAGE_PIXMAPS_DIR)
-
-dnl Subst PACKAGE_ICON_DIR.
-NO_PREFIX_PACKAGE_ICON_DIR="${packageicondir}"
-AC_SUBST(NO_PREFIX_PACKAGE_ICON_DIR)
-PACKAGE_ICON_DIR="${packageprefix}/${packageicondir}"
-AC_SUBST(PACKAGE_ICON_DIR)
-
-dnl Subst PACKAGE_HELP_DIR.
-NO_PREFIX_PACKAGE_HELP_DIR="${packagehelpdir}"
-AC_SUBST(NO_PREFIX_PACKAGE_HELP_DIR)
-PACKAGE_HELP_DIR="${packageprefix}/${packagehelpdir}"
-AC_SUBST(PACKAGE_HELP_DIR)
-
-dnl Subst PACKAGE_MENU_DIR.
-NO_PREFIX_PACKAGE_MENU_DIR="${packagemenudir}"
-AC_SUBST(NO_PREFIX_PACKAGE_MENU_DIR)
-PACKAGE_MENU_DIR="${packageprefix}/${packagemenudir}"
-AC_SUBST(PACKAGE_MENU_DIR)
-
-dnl AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${packageprefix}/${packagedatadir}")
-dnl AC_DEFINE_UNQUOTED(PACKAGE_DOC_DIR, "${packageprefix}/${packagedocdir}")
-dnl AC_DEFINE_UNQUOTED(PACKAGE_PIXMAPS_DIR, "${packageprefix}/${packagepixmapsdir}")
-dnl AC_DEFINE_UNQUOTED(PACKAGE_HELP_DIR, "${packageprefix}/${packagehelpdir}")
-dnl AC_DEFINE_UNQUOTED(PACKAGE_MENU_DIR, "${packageprefix}/${packagemenudir}")
-dnl AC_DEFINE_UNQUOTED(PACKAGE_SOURCE_DIR, "${packagesrcdir}")
-
-
-AC_DEFINE([PACKAGE_DATA_DIR], [share], [Package data dir])
-AC_DEFINE([PACKAGE_DOC_DIR], [doc], [Package doc dir])
-AC_DEFINE([PACKAGE_HELP_DIR], [doc], [Help])
-AC_DEFINE([PACKAGE_MENU_DIR], [], [Menu])
-AC_DEFINE([PACKAGE_ICON_DIR], [icons], [Icons])
-AC_DEFINE([PACKAGE_SRC_DIR], [src], [Source])
-
-
 AC_CONFIG_FILES([
 Makefile
 po/Makefile.in
@@ -254,4 +170,3 @@ icons/Makefile
 test/Makefile
 ])
 AC_OUTPUT
-
diff --git a/icons/Makefile.am b/icons/Makefile.am
index 9588562..0b17692 100644
--- a/icons/Makefile.am
+++ b/icons/Makefile.am
@@ -4,7 +4,7 @@
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
 
-lingot_iconsdir = $(prefix)/@NO_PREFIX_PACKAGE_ICON_DIR@
+lingot_iconsdir = $(datadir)/icons/hicolor/scalable/apps
 
 lingot_icons_DATA = \
 	org.nongnu.lingot.svg
From 0d65c3c71000a1f234288af4d703af888d569fd1 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Sat, 14 Jul 2018 09:07:52 +0200
Subject: [PATCH] Stop generating the desktop file.

Drop the absolute path, and allow the icon to be found dynamically.
This makes adding bitmaps additionally to the SVG easier later as well.
---
 configure.ac                                              | 1 -
 org.nongnu.lingot.desktop.in => org.nongnu.lingot.desktop | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)
 rename org.nongnu.lingot.desktop.in => org.nongnu.lingot.desktop (86%)

diff --git a/configure.ac b/configure.ac
index 5f1ea03..08f208f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -252,7 +252,6 @@ po/Makefile.in
 src/Makefile
 icons/Makefile
 test/Makefile
-org.nongnu.lingot.desktop
 ])
 AC_OUTPUT
 
diff --git a/org.nongnu.lingot.desktop.in b/org.nongnu.lingot.desktop
similarity index 86%
rename from org.nongnu.lingot.desktop.in
rename to org.nongnu.lingot.desktop
index 76842ce..8b3524f 100644
--- a/org.nongnu.lingot.desktop.in
+++ b/org.nongnu.lingot.desktop
@@ -6,7 +6,7 @@ Comment=LINGOT Is Not a Guitar-Only Tuner
 Comment[cs]=(Nejen kytarová) ladička
 Comment[fr]=Un accordeur de guitare, mais pas que
 Exec=lingot
-Icon=@PACKAGE_ICON_DIR@/org.nongnu.lingot.svg
+Icon=org.nongnu.lingot.svg
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;AudioVideo;
From 29691022bae3180cc021ad7b97023dd483be19de Mon Sep 17 00:00:00 2001
From: Iban Cereijo <ibancg@users.noreply.github.com>
Date: Wed, 18 Jul 2018 13:59:25 +0200
Subject: [PATCH] Putting the desktop file in the distributed files.

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index d61162f..d359df3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,6 @@
 lingotdocdir = ${docdir}
 lingotdoc_DATA = \
 	README \
-	COPYING \
 	AUTHORS \
 	ChangeLog \
 	NEWS \
@@ -37,6 +37,7 @@ dist_man_MANS = lingot.1
 
 EXTRA_DIST = \
 	$(lingotdoc_DATA) \
+	$(gnomemenu_DATA) \
 	$(appdata_DATA) \
 	build-aux/gitlog-to-changelog \
 	bootstrap
